community/kernel/src/main/java/org/neo4j/kernel/impl/store/StoreType.java

```diff
@@ -227,15 +228,28 @@ void close( NeoStores me, Object object )
      */
     public static StoreType typeOf( String storeFileName )
     {
+        final Optional<StoreType> optional = isStoreType( storeFileName );
+        return optional.orElseThrow(
+                () -> new IllegalArgumentException( "No enum constant for " + storeFileName + " file." ) );
+    }
+
+    /**
+     * Determine if a file name corresponds to any store type.
+     *
+     * @param fileName - name of the file to check for corresponding store type
+     * @return {@link Optional} wrapping store type, or empty optional if no store type corresponds to file name
+     */
+    public static Optional<StoreType> isStoreType( String fileName )
```

Comment:
Maybe this should be called `findStoreType` or `determineStoreType`, or `guessStoreType`.